icu4c/source/test/cintltst/cloctst.c

```diff
@@ -6018,6 +6019,47 @@ static void TestToLanguageTag(void) {
     }
 }
 
+static void TestBug20132(void) {
+    char langtag[256];
+    UErrorCode status;
+    int32_t len;
+
+    static const char inloc[] = "C";
+    static const char expected[] = "en-US-u-va-posix";
+    static const int32_t expected_len = sizeof expected - 1;
+
+    /* Before ICU-20132 was fixed, calling uloc_toLanguageTag() with a too small
+     * buffer would not immediately return the buffer size actually needed, but
+     * instead require several iterations before getting the correct size. */
+
+    status = U_ZERO_ERROR;
+    len = uloc_toLanguageTag(inloc, langtag, 1, FALSE, &status);
+
+    if (U_FAILURE(status) && status != U_BUFFER_OVERFLOW_ERROR) {
+        log_data_err("Error returned by uloc_toLanguageTag for locale id [%s] - error: %s Are you missing data?\n",
+            inloc, u_errorName(status));
+    }
+
+    if (len != expected_len) {
```

Comment:
Optional: Use assertXyZ() from intltest.h?

Reply:
I don't think it's possible to use that in cintltst. Is it?

Reply:
Oh, cintltst. Right.

We need to call each C API function (and use each [type of] C constant/macro/...) at least once from cintltst, to make sure that it works from pure C code. However, for functional testing, I always recommend to do that from intltest because it's more productive there to write readable and useful test code.